lib/p5.play.js

```diff
@@ -1241,6 +1241,36 @@ function Sprite(pInst, _x, _y, _w, _h) {
   */
   this.animation = undefined;
 
+  /**
+   * Swept collider oriented along the current velocity vector, extending to
+   * cover the old and new positions of the sprite.
+   *
+   * The corners of the swept collider will extend beyond the actual swept
+   * shape, but it should be sufficient for broad-phase detection of collision
+   * candidates.
+   *
+   * Note that this collider will have no dimensions if the source sprite has no
+   * velocity.
+   */
+  this._sweptCollider = new p5.OrientedBoundingBoxCollider();
+
+  /**
+   * If the sprite is moving, use the swept collider. Otherwise use the actual
+   * collider.
+   */
+  this._getBroadPhaseCollider = function() {
+    return (this.velocity.magSq() > 0) ? this._sweptCollider : this.collider;
+  };
```

Comment:
I think there are some weird edge cases here, due to `Sprite` already having internal consistency problems.  For example, if your swept collider is generated with a zero velocity during `Sprite.update` and mid-frame a `bounce` collision gives your sprite a nonzero velocity, suddenly you're using the swept collider instead of the normal collider (though they should be similar).  The reverse can happen too: You may generate a swept collider with a nonzero velocity, stop mid-frame, and end up using the shape's real collider.

In general this is probably a non-issue, but it gets a little weird when the actual collider is something other than an OBB; even at zero velocity, sometimes you'd be checking vs. a circle, and sometimes vs and OBB around the circle.

An alternative is to always use the swept collider, and have the v=0 special case in `update` that makes it match the collider.  I think there's a very small benefit to doing the `x*x+y*y > 0` work once per colllider per frame instead of once per sprite-sprite collision check, and maybe a larger benefit to having fewer edge cases.  A possible downside is that you may instead need to special-case updating a swept collider when a sprite hits something mid-frame, as changing the velocity does now.

Thoughts?